src/tests/functional.spec.jsx

```diff
@@ -110,4 +110,22 @@ describe('functional tests', () => {
     assertSheet(sheet)
     wrapper.unmount()
   })
+
+  it('should update all dynamic props', () => {
+    const Button = styled('button')({
+      padding: props => (props.spaced ? 10 : 0),
+
+      '@media screen': {
+        margin: props => (props.spaced ? 10 : 0)
+      }
+    })
+
+    const wrapper = mount(<Button spaced />)
+
+    expect(styled.mountSheet().toString()).toMatchSnapshot()
```

Comment:
we should use `assertSheet` here. but I checked that `sheet.renderer.getRules()` returns empty `media query` rule for function values in this case like:

```
.button-1-id{padding:10px;}@mediascreen{.button-1-id{}}
```

for static values it's ok